src/fabric/src/fabric_streams.erl

```diff
@@ -50,6 +50,19 @@ start(Workers0, Keypos, StartFun, Replacements) ->
     end.
 
 
+cleanup(Workers) ->
+    % Stop the auxiliary cleaner process as we got to the point where cleanup
+    % happesn in the regular fashion so we don't want to send 2x the number kill
+    % messages
+    case get(?WORKER_CLEANER) of
+        CleanerPid when is_pid(CleanerPid) ->
+            exit(CleanerPid, kill);
```

Comment:
You should delete `?WORKER_CLEANER` from the process dict here because you're gating on its presence in `spawn_worker_cleaner/2` so if a single process ever makes multiple fabric calls the second won't have the cleaner setup correctly.